src/strategy/dungeons/wotlk/forgeofsouls/FosTriggers.cpp

```diff
@@ -0,0 +1,44 @@
+#include "FosTriggers.h"
+#include "Playerbots.h"
+#include "AiObject.h"
+#include "AiObjectContext.h"
+
+bool MoveFromBronjahmTrigger::IsActive()
+{
+    Unit* boss = nullptr;
+    boss = AI_VALUE2(Unit*, "find target", "bronjahm");
+    if (boss && boss->HasUnitState(UNIT_STATE_CASTING))
+    {
+        if (boss->FindCurrentSpellBySpellId(SPELL_CORRUPT_SOUL))
+            return true;
+    }
+    return false;
+}
+
+bool SwitchToSoulFragment::IsActive()
+{
+    Unit* fragment = nullptr;
+    GuidVector targets = AI_VALUE(GuidVector, "possible targets no los");
+
+    for (auto i = targets.begin(); i != targets.end(); ++i)
+    {
+        Unit* unit = botAI->GetUnit(*i);
+        if (unit && unit->GetEntry() == NPC_CORRUPTED_SOUL_FRAGMENT)
+        {
+            return true;
+        }
+    }
+
+    return false;
+    
+}
+
+bool BronjahmPositionTrigger::IsActive()
```

Comment:
This check can be reduced to a single line:
`return bool(AI_VALUE2(Unit*, "find target", "bronjahm");`

Tank checks can also be added here, or this trigger scrapped entirely - this is not really a position trigger, it's just checking for the boss so it should probably be renamed and/or merged into something else

Reply:
I tried using only the move command to position the tank to a specified location. However, when the adds appear, the tank moves to the adds to hold aggro and then moves back to the specified position. At this point, the boss follows the tank, reducing the distance between the adds and the boss. Therefore, I attempted to remove the tank assist strategy during the fight.
However, after removing the tank assist strategy, the tank just stands still without taking any action, so I used the attack command to make the tank attack only the boss. I know this isn't a perfect solution. Is there any other way to solve this issue?

Meanwhile, To solve the following issue, I combined the tanktarget and tankattack actions, which is why the move command is used in the AttackAction.

> These should probably not be queued like this, and tank target is not really a necessary action, you can do dps + tank + heal targeting in a single action
https://github.com/liyunfan1223/mod-playerbots/pull/672#discussion_r1827113840